qcfractal/interface/collections/dataset.py

```diff
@@ -521,44 +582,84 @@ def _default_parameters(self,
+
+        molecules = []
+        molecule_ids = list(set(indexer.values()))
+        for i in range(0, len(molecule_ids), self.client.query_limit):
+            molecules.extend(self.client.query_molecules(id=molecule_ids[i:i + self.client.query_limit]))
+
+        molecules = pd.DataFrame.from_dict([{"molecule_id": x.id, "molecule": x} for x in molecules])
+
+        df = pd.DataFrame.from_dict(indexer, orient="index", columns=["molecule_id"])
+        df.reset_index(inplace=True)
+
+        # Outer join on left to merge duplicate molecules
+        df = df.merge(molecules, how="left", on="molecule_id")
+        df.set_index("index", inplace=True)
+        df.drop("molecule_id", axis=1, inplace=True)
+
+        return df
+
+    def _get_records(self,
+                     indexer: Dict[str, 'ObjectId'],
+                     query: Dict[str, Any],
+                     projection: Optional[Dict[str, bool]] = None,
+                     merge: bool = False) -> 'pd.Series':
         """
         Runs a query based on an indexer which is index : molecule_id
 
         Parameters
         ----------
-        indexer : str
-            The primary index of the query
+        indexer : Dict[str, 'ObjectId']
+            A key/value index of molecules to query
         query : Dict[str, Any]
             A results query
-        field : str, optional
-            The field to pull from the ResultRecords
-        scale : str, optional
-            The scale of the computation
+        projection : Optional[Dict[str, bool]], optional
+            Description
+        merge : bool, optional
+            Sum compound queries together, useful for mixing results
 
         Returns
         -------
         Series
             A Series of the data results
         """
+        self._check_client()
         self._check_state()
 
-        field = field.lower()
-
         ret = []
         for query_set in composition_planner(**query):
 
             # Set the index to remove duplicates
             molecules = list(set(indexer.values()))
-            query_set["projection"] = {"molecule": True, field: True}
+            if projection:
+                proj = {k.lower(): v for k, v in projection.items()}
+                proj["molecule"] = True
+                query_set["projection"] = proj
 
             # Chunk up the queries
             records = []
             for i in range(0, len(molecules), self.client.query_limit):
                 query_set["molecule"] = molecules[i:i + self.client.query_limit]
                 records.extend(self.client.query_results(**query_set))
 
-            records = pd.DataFrame(records, columns=["molecule", field])
+            if projection is None:
+                records = [{"molecule" : x.molecule, "record": x} for x in records]
+
+            records = pd.DataFrame.from_dict(records)
 
             df = pd.DataFrame.from_dict(indexer, orient="index", columns=["molecule"])
             df.reset_index(inplace=True)
```

Comment:
This throws `KeyError: 'molecule'` if no records are found. e.g.:
`ds.get_records(program="psi4", method="baloney")`
`ds.get_records(program="psi4",method='b3lyp-d3bj',subset=['baloney']) # 'b3lyp-d3bj' is a valid method for this ds`